### PR TITLE
[Logs]: refactored tailer tests

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Fixed flaky tests from the file scanner and the file tailer and created an abstraction to provide file offset to tailers

### CODE REVIEW

File: pkg/logs/auditor/auditor.go [hunk 1]
```diff
@@ -24,6 +24,12 @@ const defaultFlushPeriod = 1 * time.Second
 const defaultCleanupPeriod = 300 * time.Second
 const defaultTTL = 23 * time.Hour
 
+// FileOffsetStorage provides the last offset recorded by the agent after successfully sent data coming
+// from the file matching identifier
+type FileOffsetStorage interface {
```
Comment (tmichelet) [inline pkg/logs/auditor/auditor.go]: This is not ideal, the auditor shouldn't know about different sources. I'd rather slightly change the tailer

**Review by tmichelet — CHANGES_REQUESTED** [review]
Tests look good, but let's avoid updating the auditor (or at least let's not make it aware of inputs)

File: pkg/logs/auditor/auditor.go (hunk 1 above)
Comment (ajacquemot) [reply pkg/logs/auditor/auditor.go]: Agreed, I changed this.

**Review by tmichelet — APPROVED** [review]
thanks

**Pull request merged**